nautobot/ipam/models.py

```diff
@@ -425,18 +453,120 @@ def family(self):
             return self.prefix.version
         return None
 
-    def get_duplicates(self):
-        return Prefix.objects.net_equals(self.prefix).filter(vrf=self.vrf).exclude(pk=self.pk)
+    def reparent_subnets(self):
+        """
+        Determine list of child nodes and set the parent to self.
+        """
+        query = Prefix.objects.select_for_update().filter(
+            ~models.Q(id=self.id),  # Don't include yourself...
+            parent_id=self.parent_id,
+            prefix_length__gt=self.prefix_length,
+            ip_version=self.ip_version,
+            network__gte=self.network,
+            broadcast__lte=self.broadcast,
+            # namespace=self.namespace,
+        )
+
+        query.update(parent=self)
+
+    def supernets(self, direct=False, discover_mode=False, for_update=False):
+        """Return supernets of this Prefix."""
+        query = Prefix.objects.all()
+
+        if self.parent is None and not discover_mode:
+            return query.none()
+
+        if discover_mode and direct:
+            raise ValidationError("Direct is incompatible with discover_mode.")
+
+        if for_update:
+            query = query.select_for_update()
+
+        if direct:
+            return query.filter(id=self.parent.id)
+
+        return query.filter(
+            ip_version=self.ip_version,
+            prefix_length__lt=self.prefix_length,
+            network__lte=self.network,
+            broadcast__gte=self.broadcast,
+            # namespace=self.namespace,
+        )
+
+    def subnets(self, direct=False, for_update=False):
+        """Return subnets of this Prefix."""
+        query = Prefix.objects.all()
+
+        if for_update:
+            query = query.select_for_update()
+
+        if direct:
+            return query.filter(parent_id=self.id)
+
+        return query.filter(
+            ip_version=self.ip_version,
+            prefix_length__gt=self.prefix_length,
+            network__gte=self.network,
+            broadcast__lte=self.broadcast,
+            # namespace=self.namespace,
+        )
+
+    def is_child_node(self):
+        """
+        Returns whether I am a child node.
+        """
+        return self.parent is not None
+
+    def is_leaf_node(self):
+        """
+        Returns whether I am leaf node (no children).
+        """
+        return not self.child_prefixes.exists()
+
+    def is_root_node(self):
+        """
+        Returns whether I am a root node (no parent).
+        """
+        return self.parent is None
+
+    def get_ancestors(self, ascending=False):
```

Comment:
Would be good to have the a similar API signature as our `tree-queries` models, in which case this would be `def ancestors(self, include_self=False)` (possibly still including `ascending=False` as an additional optional kwarg, though that's another one I'm not sure I see the need for at this time.)

Reply:
Yeah, I'll add that.

Reply:
I thought about that, but given the existing supernets algorithm naturally excludes `self` because of `prefix_length__lt`. We could change it to `prefix_length__lte` (note the trailing `e`) and then have `include_self=False` work.